test/gurka/test_time_dependent_restrictions.cc

```diff
@@ -0,0 +1,56 @@
+#include "gurka.h"
+#include "proto/api.pb.h"
+
+#include <gtest/gtest.h>
+
+using namespace valhalla;
+
+class TimeDependentRestrictions : public ::testing::Test {
+protected:
+  static gurka::map map;
+
+  static void SetUpTestSuite() {
+    const std::string ascii_map = R"(
+          D----B----A
+          |    |
+          |    |
+          E----C----F
+     )";
+
+    const gurka::ways ways = {
+        {"AB", {{"highway", "motorway"}}}, {"BC", {{"highway", "motorway"}}},
+        {"BD", {{"highway", "motorway"}}}, {"CF", {{"highway", "motorway"}}},
+        {"DE", {{"highway", "motorway"}}}, {"EC", {{"highway", "motorway"}}},
+    };
+
+    const auto layout = gurka::detail::map_to_coordinates(ascii_map, 10);
+    const gurka::relations relations = {{{
+                                             {gurka::way_member, "AB", "from"},
+                                             {gurka::way_member, "BC", "to"},
+                                             {gurka::node_member, "B", "via"},
+                                         },
+                                         {{"type", "restriction"},
+                                          {"restriction", "no_left_turn"},
+                                          {"hour_on", "07:00"},
+                                          {"hour_off", "19:00"}}}};
+    map =
+        gurka::buildtiles(layout, ways, {}, relations, "test/data/gurka_time_dependent_restrictions",
+                          {{"mjolnir.timezone", VALHALLA_BUILD_DIR "test/data/tz.sqlite"}});
+  }
+};
+
+gurka::map TimeDependentRestrictions::map = {};
+
+TEST_F(TimeDependentRestrictions, ActiveHourRestriction) {
+  auto result =
+      gurka::do_action(valhalla::Options::route, map, {"A", "F"}, "auto",
+                       {{"/date_time/type", "1"}, {"/date_time/value", "2021-04-02T14:00"}});
```

Comment:
Can we test it for other `date_time` types (0,2 & 3)? These types invoke different routing algorithms so it'll be good to test the timed TRs here in case it breaks in one algo and not another.

You can use [parametrized tests](https://github.com/google/googletest/blob/master/docs/advanced.md#value-parameterized-tests) to pass each of the `date_time` types.

Reply:
If any of them are failing, they can be flagged with a `DISABLED_` prefix - at least then we'll get notice that they're not fixed yet, and we could add a CI step to try non-fatally running disabled tests :

https://google.github.io/googletest/advanced.html#temporarily-disabling-tests